rpc-rs/src/provider.rs

```diff
@@ -682,40 +710,43 @@ impl HostBridge {
-                let arg = HealthCheckRequest {};
-                let resp = match provider.health_request(&arg).await {
-                    Ok(resp) => resp,
-                    Err(e) => {
-                        error!("error generating health check response: {}", &e.to_string());
-                        HealthCheckResponse {
-                            healthy: false,
-                            message: Some(e.to_string()),
+        tokio::spawn(
+            async move {
+                while let Some(msg) = sub.next().await {
+                    // placeholder arg
+                    let arg = HealthCheckRequest {};
+                    let resp = match provider.health_request(&arg).await {
+                        Ok(resp) => resp,
+                        Err(e) => {
+                            error!(error = %e, "error generating health check response");
+                            HealthCheckResponse {
+                                healthy: false,
+                                message: Some(e.to_string()),
+                            }
                         }
-                    }
-                };
-                let buf = if this.host_data.is_test() {
-                    Ok(serde_json::to_vec(&resp).unwrap())
-                } else {
-                    serialize(&resp)
-                };
-                match buf {
-                    Ok(t) => {
-                        if let Some(reply_to) = msg.reply.as_ref() {
-                            if let Err(e) = this.rpc_client().publish(reply_to, &t).await {
-                                error!("failed sending health check response: {}", e.to_string());
+                    };
+                    let buf = if this.host_data.is_test() {
+                        Ok(serde_json::to_vec(&resp).unwrap())
+                    } else {
+                        serialize(&resp)
+                    };
+                    match buf {
+                        Ok(t) => {
+                            if let Some(reply_to) = msg.reply.as_ref() {
+                                if let Err(e) = this.rpc_client().publish(reply_to, &t).await {
+                                    error!(error = %e, "failed sending health check response");
+                                }
                             }
                         }
-                    }
-                    Err(e) => {
-                        // extremely unlikely that InvocationResponse would fail to serialize
-                        error!("failed serializing HealthCheckResponse: {}", e.to_string());
+                        Err(e) => {
+                            // extremely unlikely that InvocationResponse would fail to serialize
+                            error!(error = %e, "failed serializing HealthCheckResponse");
+                        }
                     }
                 }
             }
-        });
+            .instrument(tracing::trace_span!("subscribe_health", %topic)),
```

Comment:
I'm not sure it makes sense for this to be a span - do we want all health checks to be in the same span?